### ISSUE TITLE
Storage: This doesn't match our existing project ID: hello-rightscale002

### ISSUE THREAD

**Issue by papudesi** [GoogleCloudPlatform/google-cloud-python#4447]
https://github.com/GoogleCloudPlatform/google-cloud-python/blob/4b107613b0fcf139569504e9d44fa4a0793b4fd0/storage/google/cloud/storage/notification.py#L31

**Comment by chemelnucfin** [comment]
I think the project id specifications have changed.  I will try to find out more about what they should be and adjust the regex accordingly.

**Comment by chemelnucfin** [comment]
Hi papudesi,
  dhermes suggested a temporary fix while I try to find the actual project id.  I will leave this issue open until I can find the actual requirements.
```
import re

from google.cloud.storage import notification


notification._PROJECT_PATTERN = r'(?P<project>[a-z]+-[a-z]+\d+)'
notification._TOPIC_REF_PATTERN = notification._TOPIC_REF_FMT.format(
    notification._PROJECT_PATTERN,
    notification._TOPIC_NAME_PATTERN)
notification._TOPIC_REF_RE = re.compile(notification._TOPIC_REF_PATTERN)`
```

**Comment by hrfuller** [comment]
Also running into this with an existing project id of the form `my-product`. The suggested solution in #4467 wouldn't work for it anyway. I will also monkey patch the regex.

**Comment by tseaver** [comment]
The [resource manager API docs](https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects#Project.FIELDS.project_id) say:

> The unique, user-assigned ID of the Project. It must be 6 to 30 lowercase letters, digits, or hyphens. It must start with a letter. Trailing hyphens are prohibited.

**Comment by dhermes** [comment]
@tseaver Nice find! (I assume you are implicitly saying "The current regex doesn't match that.")

**Comment by chemelnucfin** [comment]
Thanks!  I'll fix it.

**Comment by tseaver** [comment]
The current regex insists on three hyphen-separated tokens, the first two of which must be alpha-only, and the last of which must be numeric-only.  A regex which fits the spec I quoted would be:

```python
_PROJECT_PATTERN = r'(?P<project>[a-z0-9-]{5,29}[a-z0-9])'
```

**Comment by tseaver** [comment]
FTR, [pythex](https://pythex.org/) is a cool way to test / debug regexes.

**Comment by chemelnucfin** [comment]
Closed via #4543
